loopia_test.go

```diff
@@ -0,0 +1,51 @@
+package loopia
+
+import (
+  "fmt"
+  "net/http"
+  "net/http/httptest"
+  "testing"
+  "io/ioutil"
+
+  "github.com/stretchr/testify/assert"
+)
+
+var (
+    // mux is the HTTP request multiplexer used with the test server.
+    mux *http.ServeMux
+
+    // client is the API client being tested
+    client *API
+
+    // server is a test HTTP server used to provide mock API responses
+    server *httptest.Server
+)
+
+func setup() {
```

Comment:
If you don't want to use global variables you can make this function take a `http.HandlerFunc` as an argument. Then pass that to `httptest.NewServer` and skip the mux.

Reply:
And instead call `server.HandleFunc`?